src/test.js

```diff
@@ -677,8 +709,10 @@ export function internalStop( test ) {
 
 // Forcefully release all processing holds.
 function internalRecover( test ) {
-	test.semaphore = 0;
-	internalStart( test );
+	if ( config.blocking ) {
```

Comment:
I'll revert it (and add the if check back to the one consumer) and see what happens.